spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/batch/JobExecutionExitCodeGenerator.java

```diff
@@ -31,7 +31,7 @@
  */
 public class JobExecutionExitCodeGenerator implements ApplicationListener<JobExecutionEvent>, ExitCodeGenerator {
 
-	private final List<JobExecution> executions = new ArrayList<>();
```

Comment:
👍 This could also break with a single-threaded event multi-caster if `getExitCode()` is called one one thread when another thread is delivering an event.